napari/_qt/qt_plugin_report.py

```diff
@@ -0,0 +1,186 @@
+"""Provides a QtPluginErrReporter that allows the user report plugin errors.
+"""
+import webbrowser
+from typing import Optional
+
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QGuiApplication
+from qtpy.QtWidgets import (
+    QComboBox,
+    QDialog,
+    QHBoxLayout,
+    QLabel,
+    QPushButton,
+    QTextEdit,
+    QVBoxLayout,
+    QWidget,
+)
+
+from ..plugins.exceptions import (
+    PLUGIN_ERRORS,
+    fetch_module_metadata,
+    format_exceptions,
+)
+
+
+class QtPluginErrReporter(QDialog):
+    """Dialog that allows users to review and report PluginError tracebacks.
+
+    Parameters
+    ----------
+    parent : QWidget, optional
+        Optional parent widget for this widget.
+    initial_plugin : str, optional
+        If provided, errors from ``initial_plugin`` will be shown when the
+        dialog is created, by default None
+
+    Attributes
+    ----------
+    text_area : qtpy.QtWidgets.QTextEdit
+        The text area where traceback information will be shown.
+    plugin_combo : qtpy.QtWidgets.QComboBox
+        The dropdown menu used to select the current plugin
+    github_button : qtpy.QtWidgets.QPushButton
+        A button that, when pressed, will open an issue at the current plugin's
+        github issue tracker, prepopulated with a formatted traceback.  Button
+        is only visible if a github URL is detected in the package metadata for
+        the current plugin.
+    clipboard_button : qtpy.QtWidgets.QPushButton
+        A button that, when pressed, copies the current traceback information
+        to the clipboard.  (HTML tags are removed in the copied text.)
+    plugin_meta : qtpy.QtWidgets.QLabel
+        A label that will show available plugin metadata (such as home page).
+    """
+
+    NULL_OPTION = 'select plugin... '
+
+    def __init__(
+        self,
+        parent: Optional[QWidget] = None,
+        *,
+        initial_plugin: Optional[str] = None,
+    ) -> None:
+        super().__init__(parent)
+
+        self.setWindowTitle('Recorded Plugin Exceptions')
+        self.setWindowModality(Qt.NonModal)
+        self.layout = QVBoxLayout()
+        self.layout.setSpacing(0)
+        self.layout.setContentsMargins(10, 10, 10, 10)
+        self.setLayout(self.layout)
+
+        self.text_area = QTextEdit()
+        self.text_area.setTextInteractionFlags(Qt.TextSelectableByMouse)
+        self.text_area.setMinimumWidth(360)
+
+        # Create plugin dropdown menu
+        self.plugin_combo = QComboBox()
+        self.plugin_combo.addItem(self.NULL_OPTION)
+        self.plugin_combo.addItems(list(sorted(set(PLUGIN_ERRORS))))
+        self.plugin_combo.currentTextChanged.connect(self.set_plugin)
+        self.plugin_combo.setCurrentText(self.NULL_OPTION)
+
+        # create github button (gets connected in self.set_plugin)
+        self.github_button = QPushButton('Open issue at github', self)
```

Comment:
```suggestion
        self.github_button = QPushButton('Open issue on GitHub', self)
```

Incidentally, should we make this more platform-agnostic? The developer could be using GitLab or some other issue tracker...?

Reply:
I guess, reading below (`.hide()`), that this is because each service will have a slightly different API, so it's hard for us to support all of them... Is that right? If so, I'm ok with that, but it might be interesting to allow plugin developers to provide a submit issue link template in the future.

Reply:
yep.  this is all github specific.  we will need to figure out the API for any specific server we want to support. I'll open an issue after this merges.